native/cocos/2d/renderer/Batcher2d.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include <2d/renderer/UIMeshBuffer.h>
+#include <cocos/2d/renderer/RenderDrawInfo.h>
+#include <cocos/2d/renderer/UIMeshBuffer.h>
+#include <cocos/base/TypeDef.h>
+#include <cocos/core/assets/Material.h>
+#include <cocos/renderer/gfx-base/GFXTexture.h>
+#include <cocos/renderer/gfx-base/states/GFXSampler.h>
+#include <cocos/scene/DrawBatch2D.h>
+#include <unordered_map>
+#include <vector>
+#include "base/Ptr.h"
+
+namespace cc {
+class Root;
+class Batcher2d {
+public:
+    Batcher2d();
+    explicit Batcher2d(Root* root);
+    ~Batcher2d();
```

Comment:
Does it need these functions:
- copy constructor
- assign constructor
- move constructor
- operator `=`

If not, please delete them explicitly.

Reply:
explicit is for the constructor which has a parameter
no-param constructor is not empty
so they are necessary

Reply:
to be resolved

Reply:
add ALL classes CC_DISALLOW_COPY_MOVE_ASSIGN(xxx)